src/metadata.cpp

```diff
@@ -951,6 +953,10 @@ void Metadata::drop_aggregate(const std::string& keyspace_name, const std::strin
 }
 
 void Metadata::clear_and_update_back(const VersionNumber& cassandra_version) {
+  // ASAN revealed that clear_and_update_back may race with update_partitions.
```

Comment:
I guess here we can add a mutex lock ONLY for `updating_` (no protection for `back_`): 
```
void Metadata::clear_and_update_back(const VersionNumber& cassandra_version) {
  back_.clear();
  updating_ = &back_;
}
```
Fixed:
```
void Metadata::clear_and_update_back(const VersionNumber& cassandra_version) {
  back_.clear();

  {
    ScopedMutex l(&mutex_);
    updating_ = &back_;
  }
}
```